src/Components/Assets/AssetType/ONVIFCamera.tsx

```diff
@@ -43,19 +40,21 @@ const ONVIFCamera = ({ assetId, facilityId, asset, onUpdated }: Props) => {
   const [refreshPresetsHash, setRefreshPresetsHash] = useState(
     Number(new Date())
   );
-  const dispatch = useDispatch<any>();
+  const {
+    data: facility,
+    loading,
+    refetch,
+  } = useQuery(routes.getPermittedFacility, {
+    pathParams: { id: facilityId },
+  });
   const authUser = useAuthUser();
   useEffect(() => {
-    const fetchFacility = async () => {
-      const res = await dispatch(getPermittedFacility(facilityId));
-
-      if (res.status === 200 && res.data) {
-        setFacilityMiddlewareHostname(res.data.middleware_address);
-      }
-    };
-
-    if (facilityId) fetchFacility();
-  }, [dispatch, facilityId]);
+    if (facility?.middleware_address) {
+      setFacilityMiddlewareHostname(facility.middleware_address);
+    } else {
+      () => refetch();
```

Comment:
Same here too... could lead to infinite requests if it's a valid rejection or empty middleware address.